app/services/concerns/ServiceWithId.php

```diff
@@ -0,0 +1,19 @@
+<?php
+namespace App\services\concerns;
+
+use App\lib\PDOFactory;
+use App\lib\Request;
+
+use PDO;
+
+abstract class ServiceWithId extends Service
+{
+    protected PDO $pdo;
+    
+    public function __construct(
+        protected Request $request,
+        protected string $id,
+    ) {
+        $this->pdo = PDOFactory::create();
```

Comment:
```suggestion
        parent::__construct();
```
でいい気がします。いまは一行ですけど増えたときに抽象クラスの両方するのが大変です。

Reply:
@SardineTa23 ほんとですね、修正します :pray: